client/dashboard/app/secondary-menu/index.tsx

```diff
@@ -227,81 +235,99 @@ function Help() {
 // User profile dropdown component
 function UserProfile() {
 	const { user, logout } = useAuth();
-	const { supports } = useAppContext();
-	const openCommandPalette = useOpenCommandPalette();
+	const navigate = useNavigate();
+	const { recordTracksEvent } = useAnalytics();
 	const [ isLoggingOut, setIsLoggingOut ] = useState( false );
 
+	const handleAccountItemClick = ( itemId: string ) => {
+		let route: AnyRoute | undefined;
+		switch ( itemId ) {
+			case 'billing':
+				route = billingRoute;
+				break;
+			case 'profile':
+				route = profileRoute;
+				break;
+			case 'preferences':
+				route = preferencesRoute;
+				break;
+			case 'security':
+				route = securityRoute;
+				break;
+		}
+
+		if ( ! route ) {
+			return;
+		}
+
+		navigate( { to: route.fullPath } );
+		recordTracksEvent( 'calypso_dashboard_user_profile_menu_item_click', { item_id: itemId } );
+	};
+
 	return (
-		<DropdownMenu
-			popoverProps={ {
-				placement: 'bottom-end',
-				offset: 8,
-			} }
-			label={ __( 'My profile' ) }
-			icon={
-				user.avatar_URL ? (
-					<img
-						className="dashboard-secondary-menu__avatar"
-						src={ user.avatar_URL }
-						alt={ __( 'User avatar' ) }
+		<Menu>
+			<Menu.TriggerButton
+				render={
+					<Button
+						className="dashboard-secondary-menu__item"
+						variant="tertiary"
+						icon={
+							user.avatar_URL ? (
+								<img
+									className="dashboard-secondary-menu__avatar"
+									src={ user.avatar_URL }
+									alt={ __( 'User avatar' ) }
+								/>
+							) : (
+								commentAuthorAvatar
+							)
+						}
 					/>
-				) : (
-					commentAuthorAvatar
-				)
-			}
-			toggleProps={ {
-				className: 'dashboard-secondary-menu__item',
-				variant: 'tertiary',
-			} }
-		>
-			{ ( { onClose } ) => (
-				<VStack spacing={ 0 }>
-					<VStack style={ { padding: '16px', borderBottom: '1px solid #ccc' } } spacing={ 1 }>
-						<Text>{ user.display_name }</Text>
-						<Text variant="muted">@{ user.username }</Text>
-					</VStack>
-					<MenuGroup>
-						<RouterLinkMenuItem to="/me/profile" onClick={ onClose }>
-							{ __( 'Account' ) }
-						</RouterLinkMenuItem>
-					</MenuGroup>
-					{ supports.commandPalette && (
-						<MenuGroup>
-							<MenuItem
-								onClick={ () => {
-									// First close the dropdown
-									onClose();
-									// Then open the command palette after a tiny delay
-									// to ensure the dropdown is fully closed
-									requestAnimationFrame( () => {
-										openCommandPalette();
-									} );
-								} }
-								shortcut="⌘K"
-							>
-								{ __( 'Command palette' ) }
-							</MenuItem>
-						</MenuGroup>
-					) }
-					<MenuGroup>
-						<MenuItem
-							disabled={ isLoggingOut }
-							onClick={ () => {
-								setIsLoggingOut( true );
-								logout().catch( () => setIsLoggingOut( false ) );
-							} }
-						>
+				}
+			/>
+			<Menu.Popover>
```

Comment:
The menu is a bit thin for me, can we give it a min-width?

<img width="183" height="391" alt="Image" src="https://github.com/user-attachments/assets/1a815594-ff65-4ca2-9aad-2f596e5541ec" />

Reply:
Sounds good. I added a min-width so that it's closer to the mockup 

<img width="3444" height="1926" alt="Image" src="https://github.com/user-attachments/assets/8178df3d-72c2-4c10-9a72-1cc78182c0f0" />